### ISSUE TITLE
Inconsistency between null and undefined  

### ISSUE THREAD

**Issue by asvetliakov** [typeorm/typeorm#584]
Hello. Great work.
I'm running into following issue, given this entity

```ts
class Ent {
  @PrimaryGeneratedColumn()
  id: number;

  @Column("varchar", { nullable: true });
  test: string | undefined | null;
}
```

and this values in DB:
```
1 | test
2 | NULL
```

Loading entity with id 2 gives me ```undefined``` for ```typeof ent.test```. Ok
Next i'd want to reset value for entity with id 1:

```ts
const ent = await connection.entityManager.findOne(Ent, { where: { id: 1 } });
ent.test = undefined; // Doesn't work, still "test" in DB, need ent.test = null;
await connection.entityManager.persist(ent);
```

Can you make them consistent? Either return ```null``` instead of ```undefined``` in case 1 (if the value in DB is NULL) or make assignment field to undefined to be represented by NULL in DB

**Comment by pleerock** [comment]
```ts
ent.test = undefined; // Doesn't work, still "test" in DB, need ent.test = null;
```
It's by design. If you want to set value in the database to null you must do `ent.test = null`.

What about loading - there are issues and people require to make null columns to return actual entity properties with null values - probably something I'll do in a near future. This probably will resolve this inconsistency issue you have.

**Comment by pleerock** [comment]
Okay now when you fetch columns with nulls they are returned to you. Available from `0.1.0-alpha.21`
